app/assets/javascripts/discourse/app/components/date-input.js

```diff
@@ -7,14 +7,26 @@ import { action } from "@ember/object";
 import loadScript from "discourse/lib/load-script";
 import { schedule } from "@ember/runloop";
 
+function isInputDateSupported() {
```

Comment:
Should the return value of this helper function be memoized? Otherwise we're creating an input element every time `useNativePicker` getter is called, but I don't think support for date picker changes **that often**. 😜

Reply:
This is why I used it a property: `useNativePicker` or am I misunderstanding some behavior here ?

Reply:
It's a getter, not a static property, so it's executed every time you access it. You could change it to a static one:

```diff
- get useNativePicker() {
-      return isInputDateSupported();
- },
+ useNativePicker: isInputDateSupported(),
```